### ISSUE TITLE
Please adapt to Android 12

### ISSUE THREAD

**Issue by gwc12412** [GravityBox/GravityBox#2133]
Please adapt to Android 12, thanks!

**Comment by ashly1111111** [comment]
H

**Comment by robinhugy** [comment]
+1
Please :$

**Comment by mrabhinv** [comment]
Any update on this?

**Comment by Ko-Hi-Dev** [comment]
I can give you a fun hint

![Screenshot_20220609-235738](https://user-images.githubusercontent.com/60625753/172952875-dc5134a2-9b49-492d-a7b8-29efba474c31.png)


It works somewhat

**Comment by asaf400** [comment]
Hello, Jumping in here to report that I managed to 'compile' the app for Android 12,
[I forked here](https://github.com/asaf400/GravityBox) default branch is now s (for snow cone)

> 
> However, even after installing the newly compiled apk, 
> the system framework does not respond to GravityBox..
> 
> Someone can pickup from this starting point, I might also check some logs and stuff..
> _I feel like this might not be about Android 12 now, but more to do with Zygisk Magisk and no more full Riru.._

Edit: 
Found a protection (if statement) for specific SDK version in the code, modified for valid SDK (31) and it WORKS!!! 
it might have some bugs, uploading debug build on my fork now,
USE AT YOUR OWN RISK!

Edit 2: 
Working version !
https://github.com/asaf400/GravityBox/releases/

Edit 3:
I just learned that XDA Member @Ko-Hi-Dev here has also made a build, you can chose now..
oh well, @Ko-Hi-Dev you didn't post the xda thread here, and I didn't check XDA before attempting (successfully) to take a crack at it myself :)
https://forum.xda-developers.com/t/unofficial-gravitybox-s-version.4456047/

**Comment by Ko-Hi-Dev** [comment]
> Hello, Jumping in here to report that I managed to 'compile' the app for Android 12, [I forked here](https://github.com/asaf400/GravityBox) default branch is now s (for snow cone)
> 
> > However, even after installing the newly compiled apk,
> > the system framework does not respond to GravityBox..
> > Someone can pickup from this starting point, I might also check some logs and stuff..
> > _I feel like this might not be about Android 12 now, but more to do with Zygisk Magisk and no more full Riru.._
> 
> Edit: Found a protection (if statement) for specific SDK version in the code, modified for valid SDK (31) and it WORKS!!! it might have some bugs, uploading debug build on my fork now, USE AT YOUR OWN RISK!
> 
> Edit 2: Working version ! https://github.com/asaf400/GravityBox/releases/
> 
> Edit 3: I just learned that XDA Member @Ko-Hi-Dev here has also made a build, you can chose now.. oh well, @Ko-Hi-Dev you didn't post the xda thread here, and I didn't check XDA before attempting (successfully) to take a crack at it myself :) https://forum.xda-developers.com/t/unofficial-gravitybox-s-version.4456047/



> Magisk

Yes indeed but I might not be able to take the time to look into it.. I'll try next week but no promises

**Comment by Austcool-Walker** [comment]
> Hello, Jumping in here to report that I managed to 'compile' the app for Android 12, [I forked here](https://github.com/asaf400/GravityBox) default branch is now s (for snow cone)
> 
> > However, even after installing the newly compiled apk,
> > the system framework does not respond to GravityBox..
> > Someone can pickup from this starting point, I might also check some logs and stuff..
> > _I feel like this might not be about Android 12 now, but more to do with Zygisk Magisk and no more full Riru.._
> 
> Edit: Found a protection (if statement) for specific SDK version in the code, modified for valid SDK (31) and it WORKS!!! it might have some bugs, uploading debug build on my fork now, USE AT YOUR OWN RISK!
> 
> Edit 2: Working version ! https://github.com/asaf400/GravityBox/releases/
> 
> Edit 3: I just learned that XDA Member @Ko-Hi-Dev here has also made a build, you can chose now.. oh well, @Ko-Hi-Dev you didn't post the xda thread here, and I didn't check XDA before attempting (successfully) to take a crack at it myself :) https://forum.xda-developers.com/t/unofficial-gravitybox-s-version.4456047/

thanks I'm gonna try this on EvolutionX android 12 GSI I'm running.
